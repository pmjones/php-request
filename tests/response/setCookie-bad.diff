001+ Fatal error: Uncaught TypeError: Argument 3 passed to SapiResponse::setCookie() must be of the type int, array given, called in /Users/pmjones/Code/pmjones/php-request/tests/response/setCookie-bad.php on line 3 and defined in /Users/pmjones/Code/pmjones/php-request/src/SapiResponse.php:100
002+ Stack trace:
003+ #0 /Users/pmjones/Code/pmjones/php-request/tests/response/setCookie-bad.php(3): SapiResponse->setCookie('cookie1', 'value1', Array)
004+ #1 {main}
001- Warning: SapiResponse::setCookie(): Unrecognized key 'nosuchoption' found in the options array in %ssetCookie-bad.php on line %d
002- 
003- Warning: SapiResponse::setCookie(): No valid options were found in the given array in %ssetCookie-bad.php on line %d
004- Cannot pass arguments after the options array
005+   thrown in /Users/pmjones/Code/pmjones/php-request/src/SapiResponse.php on line 100