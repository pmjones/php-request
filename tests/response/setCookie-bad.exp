Warning: SapiResponse::setCookie(): Unrecognized key 'nosuchoption' found in the options array in %ssetCookie-bad.php on line %d

Warning: SapiResponse::setCookie(): No valid options were found in the given array in %ssetCookie-bad.php on line %d
Cannot pass arguments after the options array