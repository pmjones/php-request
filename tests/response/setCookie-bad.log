
---- EXPECTED OUTPUT
Warning: SapiResponse::setCookie(): Unrecognized key 'nosuchoption' found in the options array in %ssetCookie-bad.php on line %d

Warning: SapiResponse::setCookie(): No valid options were found in the given array in %ssetCookie-bad.php on line %d
Cannot pass arguments after the options array
---- ACTUAL OUTPUT
Fatal error: Uncaught TypeError: Argument 3 passed to SapiResponse::setCookie() must be of the type int, array given, called in /Users/pmjones/Code/pmjones/php-request/tests/response/setCookie-bad.php on line 3 and defined in /Users/pmjones/Code/pmjones/php-request/src/SapiResponse.php:100
Stack trace:
#0 /Users/pmjones/Code/pmjones/php-request/tests/response/setCookie-bad.php(3): SapiResponse->setCookie('cookie1', 'value1', Array)
#1 {main}
  thrown in /Users/pmjones/Code/pmjones/php-request/src/SapiResponse.php on line 100
---- FAILED
