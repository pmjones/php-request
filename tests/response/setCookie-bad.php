<?php
$response = new SapiResponse();
$response->setCookie('cookie1', 'value1', ['nosuchoption' => true]);
try {
    $response->setCookie('cookie2', 'value2', ['expires' => 1234567890], '/path');
} catch (BadMethodCallException $e) {
    echo $e->getMessage() . PHP_EOL;
}
