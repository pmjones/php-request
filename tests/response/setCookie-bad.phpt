--TEST--
SapiResponse::setCookie-bad
--FILE--
<?php
$response = new SapiResponse();
$response->setCookie('cookie1', 'value1', ['nosuchoption' => true]);
try {
    $response->setCookie('cookie2', 'value2', ['expires' => 1234567890], '/path');
} catch (BadMethodCallException $e) {
    echo $e->getMessage() . PHP_EOL;
}
--EXPECTF--

Warning: SapiResponse::setCookie(): Unrecognized key 'nosuchoption' found in the options array in %ssetCookie-bad.php on line %d

Warning: SapiResponse::setCookie(): No valid options were found in the given array in %ssetCookie-bad.php on line %d
Cannot pass arguments after the options array
