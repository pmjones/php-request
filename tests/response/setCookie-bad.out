Fatal error: Uncaught TypeError: Argument 3 passed to SapiResponse::setCookie() must be of the type int, array given, called in /Users/pmjones/Code/pmjones/php-request/tests/response/setCookie-bad.php on line 3 and defined in /Users/pmjones/Code/pmjones/php-request/src/SapiResponse.php:100
Stack trace:
#0 /Users/pmjones/Code/pmjones/php-request/tests/response/setCookie-bad.php(3): SapiResponse->setCookie('cookie1', 'value1', Array)
#1 {main}
  thrown in /Users/pmjones/Code/pmjones/php-request/src/SapiResponse.php on line 100